pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/ControllerBasedBrokerSelector.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.client;
+
+import java.util.List;
+
+
+/**
+ * Maintains broker cache using controller APIs
+ */
+public class ControllerBasedBrokerSelector implements BrokerSelector {
+    private final UpdatableBrokerCache _brokerCache;
+
+    public ControllerBasedBrokerSelector(String scheme, String controllerHost,
+        int controllerPort, long brokerUpdateFreqInMillis) {
+        // TODO pass the broker cache type via props
+        _brokerCache = new ScheduledBrokerCache(scheme, controllerHost, controllerPort, brokerUpdateFreqInMillis);
+        _brokerCache.init();
+    }
+
+    @Override
+    public String selectBroker(String table) {
```

Comment:
Add a way to fetch broker list using tenant as well. The table name won't be available in case the selector is used in JDBC.

Reply:
This comes from the BrokerSelector interface. As of now even ZK based selector doesn't support tenant level broker selection. Maybe we can take it up separately? Support tenant based selection across all selector implementations?